archunit-integration-test/src/test/java/com/tngtech/archunit/PublicAPIRules.java

```diff
@@ -113,14 +117,31 @@ public class PublicAPIRules {
     public static final ArchRule only_entry_point_and_syntax_interfaces_should_be_public =
             classes()
                     .that().resideInAPackage("..syntax..")
-                    .and().haveNameNotMatching(".*" + ArchRuleDefinition.class.getSimpleName() + ".*")
+                    .and().haveNameNotMatching(".*RuleDefinition.*")
```

Comment:
I wonder whether `ModuleRuleDefinition.modules()` couldn't live in `ArchRuleDefinition` as well. After all, module rules are `ArchRule`s, too. For disconverability, it would certainly help to have a consistent entry point – especially since the rule structure is so similiar.

Reply:
If you prefer to keep `ModuleRuleDefinition`, this rule's description should be changed according to

Reply:
Okay, I see that `ArchRuleDefinition` lives in `lang.syntax`, whereas `ModuleRuleDefinition` should probably live in some `library` package.

Reply:
Yes, it is also consistent with `SliceRuleDefinition`. If we would add it to `ArchRuleDefinition` it would maybe make discoverability easier, but it would also violate the intended layer structure (where `lang` shouldn't depend on `library`).
If we would want something central like that then `ArchRuleDefinition` should not reside within `lang` but on a higher hierarchy layer. But as I said, we have also `Architectures` and `SliceRuleDefinition` as entry points to define `ArchRule`s within `library`, so at the moment it's at least not inconsistent...

Reply:
I thought the rule text is still fair as it is, because everything of the modules API also resides in a package `..syntax..` and that's the only thing the test checks :man_shrugging:

Reply:
My suggestion is to change
```diff
     public static final ArchRule only_entry_point_and_syntax_interfaces_should_be_public =
             classes()
                     .that().resideInAPackage("..syntax..")
                     .and().haveNameNotMatching(".*RuleDefinition.*")
                     .and().areNotInterfaces()
                     .and().areNotAnnotatedWith(Internal.class)
                     .and(are(not(onlyUsedAsPublicApiParameter())))
                     .should().notBePublic()
-                    .as(String.format(
-                            "Only %s and interfaces within the ArchUnit syntax (..syntax..) should be public",
-                            ArchRuleDefinition.class.getSimpleName()));
+                    .as("Only RuleDefinitions and interfaces within the ArchUnit syntax (..syntax..) should be public");
```
to match the change of commit `add modules API to syntax`:
```diff
     public static final ArchRule only_entry_point_and_syntax_interfaces_should_be_public =
             classes()
                     .that().resideInAPackage("..syntax..")
-                    .and().haveNameNotMatching(".*" + ArchRuleDefinition.class.getSimpleName() + ".*")
+                    .and().haveNameNotMatching(".*RuleDefinition.*")
                     .and().areNotInterfaces()
                     .and().areNotAnnotatedWith(Internal.class)
                     .should().notBePublic()
                     .as(String.format(
                             "Only %s and interfaces within the ArchUnit syntax (..syntax..) should be public",
                             ArchRuleDefinition.class.getSimpleName()));
```